### DISCUSSION TITLE
Errors building source file on windows 11 android studio or IntelliJ IDEA Community with Cygwin64 Terminal

### DISCUSSION THREAD

**Discussion by MrChota** [SagerNet/SagerNet#188] [Q&A] [answered]
hello am having trouble building the source code on my pc windows 11 , my aim is to add in some more features on to the application and improve on the user interface , if I get the code running and study it for a day or two but have failed to build the source file receiving multiple errors on both IDES. Thank you , hope to be helped and contribute to the community.

**Comment by MrChota** [comment]
Found a way out , first of all run Android studio or intellij IDEA as Administrator . Run install Cywin64/86 and use github clone not download the source code if you download the source file make sure to change the file structure name from SagerNet-Master "e.t.c" to SagerNet only then import and wait for it to build remove error that will first appear and continue all down . Make sure you have git installed not only github Desktop .

**Comment by nekohasekai** [comment, marked as answer]
https://github.com/SagerNet/SagerNet/discussions/119#discussioncomment-806257

**Reply by MrChota** [reply]
Thank very much , just to ask once more have followed the #199(comment) and have tried everything I could figure out but still getting the cant not find (import libv2ray.Libv2ray and import go.Seq) in SagerNet.kt  ) and as you mention before to quote " Understand the github action compilation process on your own. Are you not an android developer?" have tried all my best research I could get but failed if possible could you help me out thank you. The yml git action is relatively new to me .
